sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1343,4 +1343,13 @@ export class ApplicationService extends RecordDataModelService<Application> {
       )
       .getOne();
   }
+  async getApplicationByApplicationNumberStudent(
+    applicationNumber: string,
+    student: Student,
```

Comment:
I believe that we have been passing the parameters that we really needed. For instance, instead of providing the entire _Student_ object, we would be providing only the _studentId_.